Sources/Purchasing/StoreKit2/StoreKit2TransactionFetcher.swift

```diff
@@ -31,6 +31,12 @@ protocol StoreKit2TransactionFetcherType: Sendable {
     @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
     var firstVerifiedTransaction: StoreTransaction? { get async }
 
+    @available(iOS 16.0, macOS 13.0, tvOS 16.0, watchOS 9.0, *)
+    var appTransactionJWS: String? { get async }
+
+    @available(iOS 16.0, macOS 13.0, tvOS 16.0, watchOS 9.0, *)
+    func appTransactionJWS(_ completionHandler: @escaping (String?) -> Void)
```

Comment:
When are we using this? It seems we might be fine with just the async var?

Reply:
It's used in the TransactionPoster [here](https://github.com/RevenueCat/purchases-ios/blob/bc02c9afa2540df6c4ba8d4c39b5e438935ca03a/Sources/Purchasing/Purchases/TransactionPoster.swift#L110), in a spot that uses completion handlers in a non-async function.

Reply:
sorry had not seen that! in that case, would it make sense to remove the @available from `appTransactionJWS`  and move the iOS 16 availability check inside it, and just return nil for earlier versions?

Reply:
Love this! I'll do that and then the code in TransactionPoster will only need to call `postReceipt` once :D